### ISSUE TITLE
[css-fonts-4] May want different properties for font matching and variation value

### ISSUE THREAD

**Issue by upsuper** [w3c/csswg-drafts#528]
Before we have variable fonts, `font-weight` property is only used for font matching, and thus authors can map a weight to whatever font face they want regardless of the designed weight of the font.

However, with variable fonts, `font-weight` property is now used for both font matching and choosing value for variation, which means authors lose the ability describe above.

I suppose this applies to other properties involving font matching as well, e.g. `font-style`, `font-stretch`.

(I'm not sure whether it is an important issue, though, since I'm not a typical frontend developer or designer.)

cc @litherum @nattokirai

**Comment by frivoal** [comment]
I agree this is an issue. One example of a situation where this is a problem is when an author wants to use multiple fonts with different unicode coverage. The weights at which the fonts are visually a good match is not necessarily when the numeric value of font weight is the same.

With traditional fonts pairing font faces that match visually, and mapping them onto the same font weight number for usage in the stylesheet is easy.

Regardless of the syntax we end up with, we need to the same ability for variable fonts.

 - Florian

**Comment by Crissov** [comment]
@upsuper means authors can get creative with the `font-weight` *descriptor* currently. I don’t see how it affects the *property* of the same name.

**Comment by frivoal** [comment]
This comment of mine in https://github.com/w3c/csswg-drafts/issues/521#issuecomment-251553192 relates to this issue.

**Comment by litherum** [comment]
Can't `font-variation-settings` be used to set the rendering variation value as distinct from the one used for font selection? The specific axis names are listed explicitly in the spec for each font-selection property.

**Comment by frivoal** [comment]
I don't see how that would work, you cannot set different values of `font-variation-settings` for different font faces.

It seems to me that we need a level of indirection, similar to how you can do

```
@font-face {
  font-family: foo;
  font-weight: 700;
  src: url('..../foo-700.otf');
}
@font-face {
  font-family: bar;
  font-weight: 700;
  src: url('..../bar-900.otf');
}
.something {
  font-family: foo, bar;
}
```
Maybe you're using the font stack to deal with the fact that one font may be missing, or maybe you're using it due to different unicode coverage. Either way, you want to use bar-900.otf when the `font-weight` property is 700. There's a mismatch between 700 and 900, and it is intentional, because these numbers are on an arbitrary scale, so when you want to pair two fonts, you need to set up your own scale and place these two fonts somewhere on it.

How do we do the same with font variations? With the current syntax and semantics, you cannot:

```
@font-face {
  font-family: foo;
  font-weight: 500-700; /*use when the font-weight property is between 500 and 700*/
  src: url('..../foo.otf'); /* I'd like to use the 500-700 range of that font
                               when rendering, so all is good*/
}
@font-face {
  font-family: bar;
  font-weight: 500-700; /*use when the font-weight property is between 500 and 700*/
  src: url('..../bar.otf'); /* I'd like to use the 700-900 range of that font
                               when rendering because it is otherwise comparatively
                               lighter than font foo, but I cannot express it.*/
}
.something {
  font-family: foo, bar;
}
```

I suggest we enable something along these lines:

```
@font-face {
  font-family: foo;
  font-weight: 500-700;
  src: url('..../foo.otf');

  /* the above is the same as font-weight: 500-700 / 500-700; */
}
@font-face {
  font-family: bar;
  font-weight: 500-700 / 700-900;
  src: url('..../bar.otf');

  /* use when the font-weight property is between 500 and 700,
     but render it in with the 700-900 range,
     mapping values from one range to the other linearly */
}
.something {
  font-family: foo, bar;
}
```

(If you don't like slashes, we can have commas, keywords, or whatever, that's not the point)

You cannot do that with `font-variation-settings` (or any other property that does not distinguish between font faces), because you need different values per font-face on the same element.

**Comment by Crissov** [comment]
`font-weight-adjust` 😜

**Comment by frivoal** [comment]
> font-weight-adjust 😜

Last person I know to propose this was me, and I was proven wrong :)
https://lists.w3.org/Archives/Public/www-style/2015Oct/0151.html

**Comment by davelab6** [comment]
> RESOLVED: defer this issue to next level

When will css level 5 discussions begin? :)

**Comment by svgeesus** [comment]
> When will css level 5 discussions begin? :)

When most of CSS Fonts 4 is implemented in a couple of browsers, and it makes sense to start up a new level with any deferred features plus new ones so that level 4 can be stabilized.

**Comment by davelab6** [comment]
> When most of CSS Fonts 4 is implemented in a couple of browsers

How is that faring? Is there any place to easily check the status of that? :)
